main.tf

```
@@ -81,7 +103,7 @@ resource "random_uuid" "lambda_uuid" {}
 
 
 resource "aws_lambda_function" "this" {
-  filename         = data.archive_file.this.output_path
+  filename         = data.github_release.this.zipball_url.output
```

Should be using the destination from the curl

```suggestion
  filename         = "${path.module}/lambda.zip"
```

---

also you should remove `_source_code_hash` as it's not usable in this version